### PR TITLE
adds delete all, filter commands, flex opening

### PR DESCRIPTION
**Pull request by bernardotorres**
Hi Christoph, awesome package.
I wanted to clean up my list, which accumulated a thousand items since 2015 and it helped a lot.
For that, I added a couple of commands:
* `da`: delete all — from a list (like after filtering);
* `f`: filter (moved “by length” to `[l]ength`);
* `.`: to group by domains, to find clusters of URLs;

Also now it deals with empty lists.

Also made the `open` command multi-platform.

### CODE REVIEW

File: pocket_prompt.py [hunk 1]
```diff
@@ -4,15 +4,20 @@
...
+from urllib.parse import urlparse
+from starter_program import OPEN_COMMAND
 
 
 class PocketPrompt:
     valid_commands = {
         "v": "[v]iew <index>",
         "d": "[d]elete <index>",
+        "da": "[d]elete [a]ll",
+        ".": "domains",
```
Comment (cgkantidis) [inline pocket_prompt.py]: Please change the value to `[.] domains` so the user knows what should be entered at the prompt to show the domains.

File: pocket_prompt.py [hunk 2]
```diff
@@ -87,7 +96,49 @@ def display(self):
...
-    def prompt_filter(self):
+    def prompt_filter(self, match):
+        self.items = dict(
+            filter(
+                lambda pair: match in pair[1].given_url or match in pair[1].given_title,
+                self.items.items(),
+            )
+        )
+
+    def prompt_domains(self):
```
Comment (cgkantidis) [inline pocket_prompt.py:136]: The domain filtering doesn't work as expected, and there are some runtime errors as well.

Please use this as the implementation:

```python3
    def prompt_domains(self):
        domains = defaultdict(int)
        for item in self.items.values():
            domain = urlparse(item.given_url).hostname
            domains[domain] += 1
        domains = list(domains.items())

        while True:
            for idx, (domain, count) in enumerate(domains, start=1):
                print(f"{idx}. {count}: {domain}")
            try:
                domain_idx = int(input("> "))
                if 1 <= domain_idx <= len(domains):
                    filter_domain = domains[domain_idx - 1][0]
                    break
                raise ValueError
            except ValueError:
                print("Invalid index\n")
                pass

        self.items = dict(
            filter(
                lambda pair: urlparse(pair[1].given_url).hostname ==
                filter_domain,
                self.items.items(),
            ))
```

**Review by cgkantidis — CHANGES_REQUESTED** [review]
Thanks a lot Bernando for the PR!
Your change is very valuable!

I've left a few comments regarding some things I'd like you to improve before I merge your change though.

Could you please update accordingly?

File: pocket_prompt.py (hunk 2 above)
Comment (bernardotorres) [reply pocket_prompt.py:136]: Replaced. Thanks for the implementation :)

File: pocket_prompt.py (hunk 1 above)
Comment (bernardotorres) [reply pocket_prompt.py]: Done

**Comment by bernardotorres** [comment]
@christosg88 Changed according to your requests. Please review again.

**Pull request merged**